bootstrap_salt/fab_tasks.py

```diff
@@ -53,6 +55,48 @@ def find_master():
     return master
 
 
+@task
+def find_master():
+    _validate_fabric_env()
+    project_config = config.ProjectConfig(env.config,
+                                          env.environment,
+                                          env.stack_passwords)
+    cfg = project_config.config
+    try:
+        zone_name = cfg['master_zone']
+    except KeyError:
+        logging.warn("master_zone not found in config file, "
+                     "so DNS discovery of master not possible, "
+                     "falling back to AWS EC2 API.")
+        return get_master_ip()
+    master = 'master.{0}.{1}.{2}'.format(env.environment,
+                                         env.application,
+                                         zone_name)
+    return master
+
+
+def set_master_dns():
+    master_ip = get_master_ip()
+    project_config = config.ProjectConfig(env.config,
+                                          env.environment,
+                                          env.stack_passwords)
```

Comment:
Like this task in bootstrap-cfn you mean: https://github.com/ministryofjustice/bootstrap-cfn/blob/master/bootstrap_cfn/fab_tasks.py#L95

Yeah, sounds like a good idea.